### PR TITLE
check(smi): add check for existence of routes referenced by traffictarget

### PR DESCRIPTION
**Pull request by SanyaKochhar**

### CODE REVIEW

File: pkg/smi/access/v1alpha2/methods.go [hunk 1]
```diff
@@ -34,3 +39,23 @@ func doesTargetRefSrcPod(spec accessClient.TrafficTargetSpec, srcPod *corev1.Pod
...
+		routes.Add(httpRouteGroup.Name)
+	}
+	tcpRoutes, err := specClient.SpecsV1alpha3().TCPRoutes(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msgf("Error getting HTTPRouteGroups for namespace %s", namespace)
+	}
+	for _, tcpRoute := range tcpRoutes.Items {
+		routes.Add(tcpRoute.Name)
+	}
+	return routes, err
```
Comment (jaellio) [inline pkg/smi/access/v1alpha2/methods.go:60]: nit: If an err occurs for both `List` calls, the first `err` will be overwritten and only the second `err` will be returned. Most likely both errors would be the same. I don't think this needs to be addressed.

File: pkg/smi/access/routes_existence.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+		return check.runForTargetTargetV1alpha2()
+	case osm.V1Alpha3:
+		return check.runForTargetTargetV1alpha3()
+	default:
+		return outcomes.Fail{Error: fmt.Errorf(
+			"OSM Controller version could not be mapped to a TrafficTarget version. Supported versions are v0.5 through v0.9")}
+	}
+}
+
+func (check RoutesExistenceCheck) runForTargetTargetV1alpha2() outcomes.Outcome {
```
Comment (jaellio) [inline pkg/smi/access/routes_existence.go:69]:
```suggestion
func (check RoutesExistenceCheck) runForTrafficTargetV1alpha2() outcomes.Outcome {
```

File: pkg/smi/access/routes_existence.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	if !foundMatchingTarget {
+		return outcomes.Info{Diagnostics: fmt.Sprintf("No applicable Traffic Targets in namespace %s to check routes for", check.dstPod.Namespace)}
+	}
+	if len(missingRoutes) > 0 {
+		return outcomes.Fail{Error: fmt.Errorf("The following routes could not be found in the cluster: %s", strings.Join(missingRoutes, ", "))}
+	}
+	return outcomes.Pass{}
+}
+
+func (check RoutesExistenceCheck) runForTargetTargetV1alpha3() outcomes.Outcome {
```
Comment (jaellio) [inline pkg/smi/access/routes_existence.go:106]:
```suggestion
func (check RoutesExistenceCheck) runForTrafficTargetV1alpha3() outcomes.Outcome {
```

File: pkg/smi/access/routes_existence.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		return outcomes.Info{Diagnostics: fmt.Sprintf("No applicable Traffic Targets in namespace %s to check routes for", check.dstPod.Namespace)}
+	}
+	if len(missingRoutes) > 0 {
+		return outcomes.Fail{Error: fmt.Errorf("The following routes could not be found in the cluster: %s", strings.Join(missingRoutes, ", "))}
+	}
+	return outcomes.Pass{}
+}
+
+// TODO: update if supported routes change for the OSM controller
+func isRouteValid(routeKind string) bool {
```
Comment (jaellio) [inline pkg/smi/access/routes_existence.go:144]: nit:
```suggestion
func isValidRoute(routeKind string) bool {
```

**Review by jaellio — APPROVED** [review]

File: pkg/smi/access/v1alpha2/methods.go (hunk 1 above)
Comment (SanyaKochhar) [reply pkg/smi/access/v1alpha2/methods.go:60]: Yeah that's what I was thinking!

File: pkg/smi/access/routes_existence.go (hunk 2 above)
Comment (SanyaKochhar) [reply pkg/smi/access/routes_existence.go:69]: lol

**Pull request merged**